appcues/src/main/java/com/appcues/ui/ExperienceStepFormState.kt

```diff
@@ -1,39 +1,102 @@
 package com.appcues.ui
 
 import androidx.compose.runtime.mutableStateOf
+import com.appcues.data.model.ExperiencePrimitive.OptionSelectPrimitive
+import com.appcues.data.model.ExperiencePrimitive.TextInputPrimitive
+import com.appcues.ui.ExperienceStepFormItemState.OptionSelectFormItemState
+import com.appcues.ui.ExperienceStepFormItemState.TextInputFormItemState
 import java.util.UUID
 
 internal class ExperienceStepFormState {
```

Comment:
this class now maintains all the responsibility for getting and setting form state values, and creating and retaining references to form item state on demand. The primitives are thus simplified a bit and just interact with the get/set value functions.